### PR TITLE
make sure to connect after peering add

### PR DESCRIPTION
**Pull request by sinkingsugar**

### CODE REVIEW

File: shards/run-test-ipfs.edn [hunk 1]
```diff
@@ -42,18 +37,26 @@
...
     = .p2p-address
     ["swarm" "peering" "add" .p2p-address] = .ipfs-args
-    "" (Process.Run :Executable "ipfs" :Arguments .ipfs-args)
-    )
+    (Maybe
+     (->
+      "" (Process.Run :Executable "ipfs" :Arguments .ipfs-args))
+     (-> (Math.Dec .p2p-count)))
+    ["swarm" "connect" .p2p-address] = .ipfs-args-2
+    (Maybe
```
Comment (sinkingsugar) [inline shards/run-test-ipfs.edn]: We actually wanna see the failures here

**Comment by alebaffa** [comment]
For reference. 
Problem being addressed in this issue on Substrate: [Implement Bitswap protocol using notifications protocol](https://github.com/paritytech/polkadot-sdk/issues/524).

File: shards/run-test-ipfs.edn [hunk 2]
```diff
@@ -47,13 +42,12 @@
    (->
     (Log "connecting to p2p address")
     = .p2p-address
-    ["swarm" "peering" "add" .p2p-address] = .ipfs-args
+    ["swarm" "peering" "add" .p2p-address] >= .ipfs-args
     "" (Process.Run :Executable "ipfs" :Arguments .ipfs-args)
-    )
-   )
+    ["swarm" "connect" .p2p-address] > .ipfs-args
```
Comment (sinkingsugar) [inline shards/run-test-ipfs.edn:47]: Was this added on devel in the end @alebaffa ? If so close this please 🙏

Comment (alebaffa) [reply shards/run-test-ipfs.edn:47]: Yes, it was added.

**Pull request closed without merging**